plugin/pkg/scheduler/algorithm/predicates/utils.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package predicates
+
+import "k8s.io/kubernetes/pkg/labels"
+import "k8s.io/kubernetes/pkg/api"
+
+// findLabelsInSet gets as many constraints as possible from the pod itself.
+func findLabelsInSet(labelsToKeep []string, nodeSelector labels.Set) map[string]string {
+	aL := make(map[string]string)
+	for _, l := range labelsToKeep {
+		if nodeSelector.Has(l) {
+			aL[l] = nodeSelector.Get(l)
+		}
+		// No need to flag missing labels: if the length of this map < labelsToKeep,
```

Comment:
This should be function-level comment. Also, I wouldn't mention node in it (just to say that if map < labelsToKeep, than some labels are not present in the selector).